### PR TITLE
:rocket: Button Widget

### PR DESCRIPTION
**Pull request by AlexandreMichelix**
- Create button widget
- Missing width adjustement
- Missing icon graphic

### CODE REVIEW

File: lib/constants/design/colors.dart [hunk 1]
```diff
@@ -17,6 +17,23 @@ class BSColors {
...
+  Color purpleButton = const Color (0xffa452da);
+  Color darkGreyButton = const Color (0xFF808080);
+  Color blueButton = const Color(0xff3d6ad6);
+  Color redButton = const Color (0xfff14336);
+  Color greyButton = const Color (0xffa9a9a9);
+
+  Color brownBorder = const Color (0xFF873100);
+  Color purpleDarkBorder = const Color (0xff25143c);
+  Color blackBorder = const Color (0xff1c1c1c);
+  Color fbBorder = const Color (0xFF001035);
```
Comment (FlavianEng) [inline lib/constants/design/colors.dart:31]:
```suggestion
  Color facebookBorder = const Color (0xFF001035);
```

File: lib/shared/Button/button_widget.dart [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  State<ButtonComponent> createState() => _ButtonComponent();
+}
+
+class _ButtonComponent extends State<ButtonComponent> {
+
+//4 3 1 2
+  List<Color> styleOrange = [BSColors().orangeButton.darken(0.1), BSColors().orangeButton, BSColors().orangeButton.brighten(0.4), BSColors().orangeButton.brighten(0.2), BSColors().orangeText, BSColors().orangeText];
+  List<Color> stylePurple = [BSColors().purpleButton.darken(0.1), BSColors().purpleButton, BSColors().purpleButton.brighten(0.4), BSColors().purpleButton.brighten(0.2), BSColors().purpleDarkBorder, BSColors().whiteText];
+  List<Color> styleGray = [BSColors().darkGreyButton.darken(0.1), BSColors().darkGreyButton, BSColors().darkGreyButton.brighten(0.4), BSColors().darkGreyButton.brighten(0.2), BSColors().blackBorder, BSColors().whiteText];
+  List<Color> styleFb = [BSColors().blueButton.darken(0.1), BSColors().blueButton, BSColors().blueButton.brighten(0.4), BSColors().blueButton.brighten(0.2), BSColors().fbBorder, BSColors().whiteText];
```
Comment (FlavianEng) [inline lib/shared/Button/button_widget.dart:25]:
```suggestion
  List<Color> styleFacebook = [BSColors().blueButton.darken(0.1), BSColors().blueButton, BSColors().blueButton.brighten(0.4), BSColors().blueButton.brighten(0.2), BSColors().fbBorder, BSColors().whiteText];
```

File: lib/shared/Button/button_widget.dart [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          ],
+        );
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return InkWell(
+      onTap: () {
+        print('c\'est un bouton');
```
Comment (FlavianEng) [inline lib/shared/Button/button_widget.dart:88]: Does it works ? _Don't change anything, I am just curious_
```suggestion
        print("c'est un bouton");
```

**Review by FlavianEng — APPROVED** [review]

File: lib/shared/Button/button_widget.dart (hunk 3 above)
Comment (AlexandreMichelix) [reply lib/shared/Button/button_widget.dart:88]: Just for testing if the button is clicked, it displays a message in the console, no use even so

**Review by FlavianEng — APPROVED** [review]

**Review by DrakenLine — APPROVED** [review]

**Pull request merged**